build/commands/lib/test.js

```diff
@@ -190,128 +208,174 @@ const runTests = (passthroughArgs, suite, buildConfig, options) => {
+        convertJSONToXML = true
+      } else if (!config.isIOS()) {
+        runArgs.push(`--gtest_output=xml:${outputFilename}.xml`)
       }
+      fs.appendFileSync(allResultsFilePath, `${testSuite}.xml\n`)
+    }
 
-      if (config.targetOS === 'android' && !isJunitTestSuite) {
-        assert(
-          config.targetArch === 'x86'
-            || config.targetArch === 'x64'
-            || options.manual_android_test_device,
-          'Only x86 and x64 builds can be run automatically. For other builds please run test device manually and specify manual_android_test_device flag.',
+    if (config.isAndroid() && !isJunitTestSuite) {
+      assert(
+        config.targetArch === 'x86'
+          || config.targetArch === 'x64'
+          || options.manual_android_test_device,
+        'Only x86 and x64 builds can be run automatically. For other builds please run test device manually and specify manual_android_test_device flag.',
+      )
+
+      if (!options.manual_android_test_device) {
+        runArgs.push(
+          `--avd-config=tools/android/avd/proto/${options.android_test_emulator_name}.textpb`,
         )
+      }
+    }
 
-        if (!options.manual_android_test_device) {
-          runArgs.push(
-            `--avd-config=tools/android/avd/proto/${options.android_test_emulator_name}.textpb`,
-          )
-        }
+    if (config.isTeamcity) {
+      // Stdout and stderr must be separate for a test launcher.
+      runOptions.stdio = 'inherit'
+    }
+
+    let progStatus = 0
+
+    if (config.isIOS()) {
+      const outputDir = path.join(config.outputDir, `run_${testSuite}_out`)
+      if (fs.existsSync(outputDir)) {
+        fs.rmSync(outputDir, { recursive: true })
+      }
+      fs.mkdirSync(path.join(outputDir, 'iossim'), { recursive: true })
+      fs.mkdirSync(path.join(outputDir, 'output'), { recursive: true })
+
+      let xcodeBuildVersion = options.ios_xcode_build_version
+      if (xcodeBuildVersion == null) {
+        xcodeBuildVersion = util
+          .run('xcodebuild', ['-version'], { skipLogging: true })
+          .stdout.toString()
+          .trim()
+          .split(' ')
+          .pop()
       }
 
-      if (config.isTeamcity) {
-        // Stdout and stderr must be separate for a test launcher.
-        runOptions.stdio = 'inherit'
+      runArgs.push('--app', getTestBinary(Config, testSuite))
+      runArgs.push('--runtime-cache-prefix', `${outputDir}/Runtime-ios-`)
+      runArgs.push('--iossim', `${outputDir}/iossim`)
+      runArgs.push('--out-dir', `${outputDir}/output`)
+      runArgs.push('--xctest')
+      runArgs.push('--xcode-build-version', xcodeBuildVersion)
+      if (config.targetEnvironment === 'simulator') {
+        runArgs.push(
+          '--platform',
+          options.ios_simulator_platform || 'iPhone 16',
+        ) // update as needed for version
+        runArgs.push('--version', options.ios_simulator_version || '18.4') // should match ios_deployment_target
```

Comment:
I wonder what's a better way to store these fallback values. Maybe put them in `Config` so we can update the config instead of modifying `test.js` each time?

Reply:
We have a mix of defaults in commands.js and config.js. Putting them in config.js means we also need a var for each one and adds to complexity here so I wonder if standardizing on commands.js makes more sense. I think commander has a way to share common options and their associated defaults